src/uu/tail/src/follow/watch.rs

```diff
@@ -477,7 +480,7 @@ pub fn follow(mut watcher_service: WatcherService, settings: &Settings) -> UResu
 
         // If `--pid=p`, tail checks whether process p
         // is alive at least every `--sleep-interval=N` seconds
-        if settings.follow.is_some() && settings.pid != 0 && process.is_dead() {
+        if settings.follow.is_some() && observer.pid != 0 && process.is_dead() {
```

Comment:
Why this change from `settings.pid` to `observer.pid`?

Reply:
`settings.pid` only stores the initial configuration given by the user from the command line, but the observer acts upon the final value, for example after checking the pid support on the platform.

Reply:
Ah gotcha, that makes sense